### PR TITLE
Restrict merges from wide to compact parts

### PR DESCRIPTION
**Pull request by CurtizJ**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Restrict merges from wide to compact parts. In case of vertical merge it led to broken result part.

### CODE REVIEW

File: src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp [hunk 1]
```diff
@@ -79,13 +79,16 @@ void FutureMergedMutatedPart::assign(MergeTreeData::DataPartsVector parts_)
 
     size_t sum_rows = 0;
     size_t sum_bytes_uncompressed = 0;
+    MergeTreeDataPartType future_part_type;
```
Comment (nikitamikhaylov) [inline src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp:82]: Maybe better to initialize this value with UNKNOWN?

**Comment by filimonov** [comment]
> from wide to compact

Can't vertical merge happen on compact parts?

File: src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp (hunk 1 above)
Comment (CurtizJ) [reply src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp:82]: Ok. It's initialized by `UNKNOWN`  by default, but maybe it's a bit confusing.

**Comment by CurtizJ** [comment]
> > from wide to compact
> 
> Can't vertical merge happen on compact parts?

No.

**Comment by nikitamikhaylov** [comment]
Arcadia check is fixed in master and I hope this PR won't break it down again.

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**